publicodes/source/mecanisms/situation.tsx

```diff
@@ -0,0 +1,89 @@
+import { isEmpty } from 'ramda'
+import {
+	makeJsx,
+	mergeAllMissing,
+	registerEvaluationFunction
+} from '../evaluation'
+import parse from '../parse'
+
+function MecanismSituation({ explanation, nodeValue, unit }) {
+	// TODO : vue différente selon si valeur depuis la situation ou calculée
+	return makeJsx({ ...explanation.valeur, nodeValue, unit })
+}
+
+type Explanation = {
+	situationKey: string
+	valeur: any //Todo type ParsedNode;
+	situationValeur?: any
+}
+export default function parseSituation(v, context) {
```

Comment:
Cette fonction n'est pas encore utilisée ? Et pourquoi passer par un mécanisme pour gérer les valeurs de la situation ?

Reply:
Cette fonction est utilisée. Elle permet de récuperer une valeur dans la situation. Cette logique n'est pas implémentée dans 'référence' car elle correspond à une étape différente de la chaine d'évaluation. Notamment à cause des contrainte suivante : 

- Si la règle est non applicable, alors elle n'est pas considérée comme manquante
- Si un parent de la règle est non applicable, alors elle n'est pas considérée comme manquante

> Et pourquoi passer par un mécanisme pour gérer les valeurs de la situation ?

Parce que le but de cette PR est de passer uniquement par des mécanisme pour tout le processus d'évaluation. Ce qu'on appel 'mécanisme' est en fait une 'brique de base' du modèle d'évaluation. C'est l'assemblage de ces briques qui constitue le modèle d'évaluation du langage. C'est un premier pas nécessaire vers une formalisation plus facile et cohérente. 

Dans le futur, le but est de pouvoir gérer la situation comme un remplacement. Mais en attendant d'y être, on passe par cette étape intermédiaire, qui permet de bien séparer les responsabilité et d'oter la gestion de la situation de l'évaluation des références.

Reply:
Ce que je n'arrive pas à comprendre c'est comment on arrive sur un nœud `situation` dans l'AST ? `setSituation()` va modifier les `parsedDefinitions` ?

Reply:
C'est un noeud qui dit : renvoie la valeur évaluée de this.parsedSituation[situationKey] et met missingVariable à `[situationKey]: 1` si elle n'existe pas.

Peut-être faut-il changer le nom du mécanisme `situation` ? Par `nom dans la situation` ? Ou `clé dans la situation` ? Autre chose ?